conjur/init_logic.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitLogic module
+
+This module is the business logic for writing configuration information
+to the user's machine as well as fetching certificates from Conjur
+"""
+
+# Builtins
+import logging
+import sys
+import os.path
+from pathlib import Path
+
+# Third party
+import socket
+import yaml
+from OpenSSL import SSL
+from OpenSSL.crypto import FILETYPE_PEM
+
+# pylint: disable=raise-missing-from
+class InitLogic:
+    """
+    InitLogic
+
+    This class holds the business logic for population the
+    conjurrc configuration details needed to connect with Conjur
+    """
+    DEFAULT_CONFIG_FILE = os.path.expanduser(os.path.join('~', '.conjurrc'))
+    DEFAULT_PROMPT_RESPONSE=True
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def connect_to_socket(hostname, port):
+        """
+        Method for opening a socket with the Conjur server
+        """
+        try:
+            if port is None:
+                port = 443
+
+            ctx = SSL.Context(method=SSL.TLSv1_2_METHOD)
+            # We set the following to false because we want to fetch the
+            # certificates and not use them to validate yet
+            ctx.check_hostname = False
+            ctx.verify_mode = False
+
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock = SSL.Connection(context=ctx, socket=sock)
+            sock.connect((hostname, port))
+            sock.do_handshake()
+        except Exception:
+            raise Exception(f"Unable to retrieve certificate from {hostname}:{port}")
+
+        logging.info("TLS Connection established. " \
+                     "Fetching certificate from server...")
+
+        return sock
+
+    # pylint: disable=unused-variable
+    @staticmethod
+    def get_certificate(hostname, port, sock, dump_certificate_func):
+        """
+        Method for connecting to Conjur to fetch the certificate chain
+        """
+        try:
```

Comment:
Content of this try/catch seem like ssl helper util rather than core code of this class

Reply:
maybe but it's only a few lines. When would logic be a helper and when would it be a core function of a class?

Reply:
Anytime something feels independent and it spans a whole function/two/three. In this case, you have very, very ssl-specific code that can be split out and separately tested